src/metrics/representation.py

```diff
@@ -146,228 +150,225 @@ def compute(self) -> float:
             scores: The scores tensor, see representation dataset for more information, 
                 size (batch_size, total_num_queries).
+                Example of score matrix:
+                    SCORES = torch.tensor(
+                        [
+                            [0, 0, 0],
+                            [1, 0, 0],
+                            [0, 0, 0],
+                            [0, 0, 0],
+                            [0, 2, 0],
+                            [0, 0, 1],
+                            [0, 0, 2],
+                            [0, 0, 4],
+                            [0, 4, 0],
+                        ]
+                    ),
+                it has 3 queries in [0, 2, 3] row position where only zeros, in order to find relevant vector index you
+                need to look at the column, first column for first query, second column for second and so on. For query
+                0 relevant vector index would be 1 with score 1, for query 1 - relevant = [4, 8] with scores = [2, 4] 
+                respectively, for query 2 - relevant = [5, 6, 7] with scores = [1, 2 , 4] respectively.
 
         Returns:
-            q_vecs: Queries vectors, size (queries_size, embedding_size).
-            db_vecs: Database vectors, for faiss build index, size (database_size, embedding_size).
-            relevant: Array of arrays relevant indexes in database for every query request, size (queries_size, ).
-            scores: Array of scores related to queries that have at least one relevant item.
-            db_idxs: Array with all database indexes.
-            queries_idxs: Array of queries order number.
-        """
-        is_queries = queries_idxs >= 0
-        queries_idxs = queries_idxs[is_queries]
-        q_vecs = vectors[is_queries]
-        db_vecs = vectors[~is_queries]
-        db_idxs = np.arange(len(db_vecs))
-        scores = scores[~is_queries]
-        
-        relevant = []
-        empty_relevant_idxs = []
+            relevants: Array of relevant indexes in gallery data, for every query.
+            clear_scores: Array of scores with dropped query rows (rows with only zeros values).
+            gallery_idxs: Array of gallery indexes in vectors storage.
+            queries_idxs: Array of queries indexes in vectors storage.
+            clear_query_order_numbers: Array of queries order numbers without -1 elements. It size like queries_idxs,
+                so if you want to get order number of queries_idxs[i] it would be clear_query_order_numbers[i].
 
-        for idx in range(len(q_vecs)):
-            relevant_idxs = np.where(scores[:, queries_idxs[idx]] > 0.)[0]
+        Raises:
+            ValueError: If dataset has query vector without relevants.
+        """
+        is_queries = query_order_numbers >= 0
+        full_indexes = np.arange(len(query_order_numbers))
+        clear_query_order_numbers = query_order_numbers[is_queries]
+        queries_idxs = full_indexes[is_queries]
```

Comment:
Looks like `queries_idxs` is exactly equal to `query_order_numbers`

Reply:
it's equal only when dataset is has attribute `shuffle=False`